bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/WorkflowDelegationStep.java

```diff
@@ -0,0 +1,161 @@
+package com.adobe.acs.commons.workflow.process.impl;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.jcr.Session;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Properties;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.granite.workflow.WorkflowException;
+import com.adobe.granite.workflow.WorkflowSession;
+import com.adobe.granite.workflow.exec.WorkItem;
+import com.adobe.granite.workflow.exec.WorkflowData;
+import com.adobe.granite.workflow.exec.WorkflowProcess;
+import com.adobe.granite.workflow.metadata.MetaDataMap;
+import com.adobe.granite.workflow.model.WorkflowModel;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+
+
+/**
+ * 
+ * This workflow steps just calls another workflow as subworkflow.
+ * 
+ * The subworkflow is determined by looking up the correct workflow on the 
+ * resource (or its ascendents). Works on pages and assets.
+ * 
+ * In order to support multiple instances of this behaviour, the name of the property,
+ * which determines the model of the subworkflow, can be be configured as arguments on
+ * the workflow step.
+ * 
+ * Arguments should be provided as this:
+ * 
+ * workflowModelProperty=<propName>,defaultWorkflowModel=<pathToDefaultWorkflowModel>
+ * 
+ * where <propName> is the name of the property which contains the paths of the workflow
+ * models
+ * 
+ * and <pathToDefaultWorkflowModel> the path to the default Workflow Model, which is used
```

Comment:
Is this path to the `/etc/workflow/models/some-workflow` or `/etc/workflow/models/some-workflow/jcr:content/model`? The  latter right? (i always get mixed up on this)